internal/provider/models/processors/test/filter_test.go

```diff
@@ -0,0 +1,291 @@
+package processors
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	. "github.com/mezmo/terraform-provider-mezmo/internal/provider/providertest"
+)
+
+func TestFilterProcessor(t *testing.T) {
+	const cacheKey = "filter_resources"
+	SetCachedConfig(cacheKey, `
+		resource "mezmo_pipeline" "test_parent" {
+			title = "pipeline"
+		}
+		resource "mezmo_http_source" "my_source" {
+			pipeline_id = mezmo_pipeline.test_parent.id
+		}`,
+	)
+
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: testAccProtoV6ProviderFactories,
+		PreCheck:                 func() { TestPreCheck(t) },
+		Steps: []resource.TestStep{
+			// Error: pipeline_id is required
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_filter_processor" "my_processor" {
+						action = "allow"
+						conditional = {
+							expressions = [
+								{
+									field = ".status"
+									operator = "equal"
+									value_number = 200
+								}
+							]
+						}
+					}`,
+				ExpectError: regexp.MustCompile("The argument \"pipeline_id\" is required"),
+			},
+
+			// Error: `action` is required
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_filter_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						conditional = {
+							expressions = [
+								{
+									field = ".status"
+									operator = "equal"
+									value_number = 200
+								}
+							]
+						}
+					}`,
+				ExpectError: regexp.MustCompile("The argument \"action\" is required, but no definition was found"),
+			},
+			// Error: `action` must be one of the allowed values
+			{
+				Config: GetCachedConfig(cacheKey) + `
+					resource "mezmo_filter_processor" "my_processor" {
+						pipeline_id = mezmo_pipeline.test_parent.id
+						action = "unknown"
+						conditional = {
+							expressions = [
+								{
+									field = ".status"
+									operator = "equal"
+									value_number = 200
+								}
+							]
+						}
+					}`,
+				ExpectError: regexp.MustCompile("Attribute action value must be one of: \\[\"allow\" \"drop\"\\], got: \"unknown\""),
```

Comment:
tip: Use backticks for these strings and you only have to escape the bracket once:

https://github.com/mezmo/terraform-provider-mezmo/blob/e66182e255cabed1ffbf4b03642e04d9ad6903af/internal/provider/models/processors/test/metrics_tag_cardinality_limit_test.go#L108